api/billingd/service/service.go

```diff
@@ -505,10 +505,10 @@ func (s *Service) createCustomer(
 	}
 	log.Debugf("created customer %s with id %s", doc.Key, doc.CustomerID)
 
-	go s.analytics.Update(doc.Key, doc.Email, false, map[string]interface{}{
-		"parent_key":   doc.ParentKey,
-		"customer_id":  doc.CustomerID,
-		"account_type": doc.AccountType,
+	go s.analytics.Identify(doc.Key, doc.AccountType, false, doc.Email, map[string]interface{}{
```

Comment:
Just curious if you're using go routines when making client calls because of possible slow requests? If that is a real concern, maybe it's safer to move the go routine forking to the client itself just so no one forgets to make the calls using a go routine. Another thought is that we should make sure to log ERROR inside of the gRPC client code in the case of a error since we are forking off go routines and have no way of knowing about errors there. If we log ERROR, we can set up some monitoring to let us know if/when our analytics tracking is down (which is quite important if we hang features like grant fil off of it) cc @textileben

Reply:
the api requests are in their own routine anyway, since batch management is automatic over there. so just removing this routine and calling it directly.